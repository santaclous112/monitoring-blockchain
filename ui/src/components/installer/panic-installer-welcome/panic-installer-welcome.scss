@import '../../scss/mixins/centralized-container.scss';
@import '../../scss/mixins/heading.scss';

.panic-installer-welcome {

  &__container {
    padding: 5px 5% 5px 5%;

    svc-surface {
      margin: 10px;
    }
  }

  &__heading {
    @include heading-container();

    svc-label {
      font-size: 25px;
      font-weight: 900;
      margin-bottom: 25px;
    }
  }

  &__message {
    font-size: 17px;
    color: var(--ion-color-primary);
    margin-left: 2.2em;
  }

  &__list {
    list-style-type: none;    
  }

  &__list-item {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: flex-start;

    color: var(--ion-color-primary);
    font-weight: 900;
    font-size: 17px;
    margin-bottom: 12px;

    ion-icon {
      margin-right: 10px;
    }
  }

  &__list-item-subtitle {
    display: flex;
    align-items: center;
    font-weight: 900;
  }

  &__list-item-excerpt {
    font-weight: 100;
  }

  &__button {
    display: flex;
    justify-content: center;
    margin: 50px 0 0 0;

    ion-button {
      font-size: 16px;
    }
  }

}