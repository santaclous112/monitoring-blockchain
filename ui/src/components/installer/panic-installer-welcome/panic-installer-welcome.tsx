import {Component, h, Host, Prop} from '@stencil/core';
import {
  ALERTS_TEXT,
  BLOCKCHAIN_TEXT,
  CHANNELS_TEXT,
  FEEDBACK_TEXT,
  INTRO_TEXT,
  NODES_TEXT,
  REPOS_TEXT
} from "../content/welcome";
import {Router} from "stencil-router-v2";
import { dismissModal } from '@simply-vc/uikit';

@Component({
  tag: 'panic-installer-welcome',
  styleUrl: 'panic-installer-welcome.scss'
})
export class PanicInstallerWelcome {

  /**
   * Stencil Router object for page navigation.
   */
  @Prop() router: Router;

  componentWillLoad() {
    // might be necessary
    // i.e. when the node operator opens a modal and then clicks in the browser's back button
    dismissModal();
  }

  render() {
    return (
      <Host>
        <panic-header showMenu={false} />

        <svc-content-container class={"panic-installer-welcome__container"}>
          <svc-surface>
            <div class={"panic-installer-welcome__heading"}>
              <svc-icon name={"rocket"} size={"120px"} color={"primary"} />
              <svc-label>Installer Journey</svc-label>
            </div>

            <svc-text color={"primary"}>
              <p class={"panic-installer-welcome__message"}>
                {INTRO_TEXT}
              </p>
            </svc-text>

            <ul class={"panic-installer-welcome__list"}>
              <li class={"panic-installer-welcome__list-item"}>
                <div class={"panic-installer-welcome__list-item-subtitle"}>
                  <svc-icon name={"link"} size={"25px"} /> Blockchain
                </div>
                <div class={"panic-installer-welcome__list-item-excerpt"}>
                  {BLOCKCHAIN_TEXT}
                </div>
              </li>

              <li class={"panic-installer-welcome__list-item"}>
                <div class={"panic-installer-welcome__list-item-subtitle"}>
                  <svc-icon name={"call"} size={"25px"} />
                  Channels
                </div>
                <div class={"panic-installer-welcome__list-item-excerpt"}>
                  {CHANNELS_TEXT}
                </div>
              </li>

              <li class={"panic-installer-welcome__list-item"}>
                <div class={"panic-installer-welcome__list-item-subtitle"}>
                  <svc-icon name={"server"} size={"25px"} />
                  Nodes
                </div>
                <div class={"panic-installer-welcome__list-item-excerpt"}>
                  {NODES_TEXT}
                </div>
              </li>

              <li class={"panic-installer-welcome__list-item"}>
                <div class={"panic-installer-welcome__list-item-subtitle"}>
                  <svc-icon name={"code-slash"} size={"25px"} />
                  Repositories
                </div>
                <div class={"panic-installer-welcome__list-item-excerpt"}>
                  {REPOS_TEXT}
                </div>
              </li>

              <li class={"panic-installer-welcome__list-item"}>
                <div class={"panic-installer-welcome__list-item-subtitle"}>
                  <svc-icon name={"notifications-circle"} size={"25px"} />
                  Alerts
                </div>
                <div class={"panic-installer-welcome__list-item-excerpt"}>
                  {ALERTS_TEXT}
                </div>
              </li>

              <li class={"panic-installer-welcome__list-item"}>
                <div class={"panic-installer-welcome__list-item-subtitle"}>
                  <svc-icon name={"checkmark-circle"} size={"25px"} />
                  Feedback
                </div>
                <div class={"panic-installer-welcome__list-item-excerpt"}>
                  {FEEDBACK_TEXT}
                </div>
              </li>
            </ul>

            <div class={"panic-installer-welcome__button"}>
              <svc-button
                iconName={"arrow-forward-circle"}
                color={"secondary"}
                href={"/installer/sub-chain"}
              >
                Start Configuration
              </svc-button>
            </div>
          </svc-surface>
        </svc-content-container>

        <panic-footer />
      </Host>
    );
  }

}