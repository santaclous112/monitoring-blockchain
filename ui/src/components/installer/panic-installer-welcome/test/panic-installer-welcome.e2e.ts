import { newSpecPage } from '@stencil/core/testing';
import { PanicInstallerWelcome } from '../panic-installer-welcome';

describe('panic-installer-welcome', () => {
  it('renders', async () => {
      const page = await newSpecPage({
          components: [PanicInstallerWelcome],
          html: '<panic-installer-welcome></panic-installer-welcome>',
      });

      const panicInstallerWelcome = page.body.querySelector('panic-installer-welcome');
      expect(panicInstallerWelcome).toMatchSnapshot();
  });
});