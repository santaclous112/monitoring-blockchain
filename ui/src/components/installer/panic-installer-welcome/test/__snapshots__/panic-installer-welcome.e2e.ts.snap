// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`panic-installer-welcome renders 1`] = `
<panic-installer-welcome>
  <panic-header></panic-header>
  <svc-content-container class="panic-installer-welcome__container">
    <svc-surface>
      <div class="panic-installer-welcome__heading">
        <svc-icon color="primary" name="rocket" size="120px"></svc-icon>
        <svc-label>
          Installer Journey
        </svc-label>
      </div>
      <svc-text color="primary">
        <p class="panic-installer-welcome__message">
          The following steps will have PANIC configured and ready!
        </p>
      </svc-text>
      <ul class="panic-installer-welcome__list">
        <li class="panic-installer-welcome__list-item">
          <div class="panic-installer-welcome__list-item-subtitle">
            <svc-icon name="link" size="25px"></svc-icon>
            Blockchain
          </div>
          <div class="panic-installer-welcome__list-item-excerpt">
            Start by adding the blockchain software being used (i.e. Akash, Polkadot, Matic, etc) based on the framework supported by PANIC (Cosmos-SDK, Substrate or Chainlink).
          </div>
        </li>
        <li class="panic-installer-welcome__list-item">
          <div class="panic-installer-welcome__list-item-subtitle">
            <svc-icon name="call" size="25px"></svc-icon>
            Channels
          </div>
          <div class="panic-installer-welcome__list-item-excerpt">
            Then pick your preferred alerting methods, such as Telegram, Slack, Twilio and e-mail.
          </div>
        </li>
        <li class="panic-installer-welcome__list-item">
          <div class="panic-installer-welcome__list-item-subtitle">
            <svc-icon name="server" size="25px"></svc-icon>
            Nodes
          </div>
          <div class="panic-installer-welcome__list-item-excerpt">
            Select the nodes you'd want to have monitored. These can be validators, full nodes, and more.
          </div>
        </li>
        <li class="panic-installer-welcome__list-item">
          <div class="panic-installer-welcome__list-item-subtitle">
            <svc-icon name="code-slash" size="25px"></svc-icon>
            Repositories
          </div>
          <div class="panic-installer-welcome__list-item-excerpt">
            Choose whether to receive alerts when your favorite blockchain project releases a new version.
          </div>
        </li>
        <li class="panic-installer-welcome__list-item">
          <div class="panic-installer-welcome__list-item-subtitle">
            <svc-icon name="notifications-circle" size="25px"></svc-icon>
            Alerts
          </div>
          <div class="panic-installer-welcome__list-item-excerpt">
            Set which metrics and their respective thresholds should be monitored to trigger alerts.
          </div>
        </li>
        <li class="panic-installer-welcome__list-item">
          <div class="panic-installer-welcome__list-item-subtitle">
            <svc-icon name="checkmark-circle" size="25px"></svc-icon>
            Feedback
          </div>
          <div class="panic-installer-welcome__list-item-excerpt">
            You're done! The blockchain software is fully configured. You'll be able to restart the installer or visit the home page to check it out!
          </div>
        </li>
      </ul>
      <div class="panic-installer-welcome__button">
        <svc-button color="secondary" href="/installer/sub-chain" iconname="arrow-forward-circle">
          Start Configuration
        </svc-button>
      </div>
    </svc-surface>
  </svc-content-container>
  <panic-footer></panic-footer>
</panic-installer-welcome>
`;
